util/deephash/deephash.go

```diff
@@ -157,7 +157,25 @@ func HasherForType[T any]() func(*T) Sum {
 	var v *T
 	seedOnce.Do(initSeed)
 	t := reflect.TypeOf(v).Elem()
-	hash := lookupTypeHasher(t)
+	return hasherFunc[T](t, lookupTypeHasher(t))
+}
+
+// HasherForStructTypeWithFieldFilter is like HasherForType but requires that T
+// be a struct and accepts a filter function to choose which structs get hashed.
+//
+// This API only permits selecting the top layer of filtering. You can't choose
+// filter paths of fields down through structs of structs.
+func HasherForStructTypeWithFieldFilter[T any](filter func(reflect.StructField) bool) func(*T) Sum {
```

Comment:
open for alternative names if this one is too long 😂

Reply:
Could do an opt parameter on HasherForType instead

Reply:
Like 

`HasherForType[T any](opts ...HasherOpt)`

Reply:
Could. It does add 3 API symbols rather than 1, though. Do you foresee any more HasherOpts in the future?

Reply:
Nope, I'm fine with this too. Just giving an alternative

Reply:
Done.